src/libYARP_os/src/yarp/os/NetworkClock.cpp

```diff
@@ -111,7 +112,11 @@ bool NetworkClock::Private::read(ConnectionReader& reader)
 
     listMutex.lock();
     auto waiter_it = waiters->begin();
-    while (waiter_it != waiters->end()) {
+    // Update the wake-up time. In case of a time reset it closes the gap between the waiter and _time.
+    waiter_it->first = _time + (waiter_it->first - oldTime);
```

Comment:
```suggestion
    if (oldTime > _time) {
        // Update the wake-up time. In case of a time reset it closes the gap
        // between the waiter and _time.
        waiter_it->first = _time + (waiter_it->first - oldTime);
    }
```